### PR TITLE
feat(postData): roll firefox to r1137, support binary post data

### PR DESCRIPTION
**Pull request by pavelfeldman**
This has a partial fix for https://github.com/microsoft/playwright/issues/2588

### CODE REVIEW

File: src/chromium/crNetworkManager.ts [hunk 1]
```diff
@@ -342,10 +342,14 @@ class InterceptableRequest implements network.RouteDelegate {
       headers,
       method,
       url,
-      postData = null,
+      postDataEntries = null,
     } = requestPausedEvent ? requestPausedEvent.request : requestWillBeSentEvent.request;
     const type = (requestWillBeSentEvent.type || '').toLowerCase();
-    this.request = new network.Request(allowInterception ? this : null, frame, redirectedFrom, documentId, url, type, method, postData, headersObject(headers));
+    let postDataBuffer = null;
+    if (postDataEntries && postDataEntries.length && postDataEntries[0].bytes)
```
Comment (dgozman) [inline src/chromium/crNetworkManager.ts:349]: Just a single entry?

File: test/network.jest.js [hunk 2]
```diff
@@ -133,13 +133,28 @@ describe('Request.postData', function() {
     expect(request).toBeTruthy();
     expect(request.postData()).toBe('{"foo":"bar"}');
   });
+  it('should work with binary', async({page, server}) => {
+    await page.goto(server.EMPTY_PAGE);
+    server.setRoute('/post', (req, res) => res.end());
+    let request = null;
+    await page.route('/post', route => route.continue());
```
Comment (dgozman) [inline test/network.jest.js:140]: Why intercept?

**Review by dgozman — APPROVED** [review]

File: src/chromium/crNetworkManager.ts (hunk 1 above)
Comment (pavelfeldman) [reply src/chromium/crNetworkManager.ts:349]: Yes, we'll only support single entry of multi-part body for now.

File: test/network.jest.js (hunk 2 above)
Comment (pavelfeldman) [reply test/network.jest.js:140]: We update postData upon interception and it can rewrite out value...

**Pull request merged**